### PR TITLE
bittrex api example

### PR DESCRIPTION
**Pull request by mrdougwright**
This PR is a WIP, and not necessary really for the repo. Just wanted to get your feedback.

Trying to get a simple "api fetch example" working, based on the work I did with cell.js:
https://github.com/mrdougwright/bittrex-cell-testing/blob/master/bittrex.html

Note: you have to enable CORS (I use a chrome ext) to get the fetch api call to work.

I'm getting the response and listening to the update on `data`, but not sure how to update the dom afterwords with row data.

### CODE REVIEW

File: examples/bittrex.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+function apiCall () {
+  return fetch("https://bittrex.com/api/v1.1/public/getmarketsummaries", {})
+    .then(res => res.json())
+    .then(obj => obj.result)
+    .then(data => data)
+}
+
+const responseHeaders = ["Market Name", "Volume", "Bid", "Open Buy Orders", "Created"]
+const concatWord = (word) => word.split(" ").reduce((a,b) => a.concat(b))
```
Comment (jayrbolton) [inline examples/bittrex.js]: could this be `word.replace(' ', '')`?

**Comment by jayrbolton** [comment]
This looks like it's 95% there. What you want to do is turn your `bd.data` data into a dynamic collection of dom nodes. You can do this with `dom.childSync`. Something similar to:

```js
const rows = dom.childSync(
  dataRow => rowView(bd, dataRow), 
  document.createElement('tbody'),
  bd,
  'data'
)
```
`rowView` could return a `tr` element with tds inside. The `rows` variable is going to be a tbody element with a dynamic set of rows inside.

**Comment by jayrbolton** [comment]
To explain a little: the view function gets run immediately on pageload, but the ajax to get the data is delayed. So when the view function is run, the data is empty.

 In order to create a dom node that has a bunch of dynamic elements that updates efficiently as some state data changes over time, then we can use `dom.childSync`

**Comment by mrdougwright** [comment]
Sweet, thanks for the help!
Well, if you want this example (even though CORS has to be enabled to work) go ahead and merge.
Otherwise, closing PR is fine.
I had fun getting the hang of "uvu" 😉 
screen shot 2017-10-31 at 10 24 23 am

**Comment by jayrbolton** [comment]
good stuff! let me see if i can get it to work with the wikipedia api so we dont need a chrome extension to run it

Also I had no idea about `window.fetch()` -- js changes so fast

**Comment by mrdougwright** [comment]
oooh, api with no cors...sounds good.

**Comment by jayrbolton** [comment]
Moved here: https://github.com/jayrbolton/ui-system-proto-tmp/pull/3/

**Pull request closed without merging**